appserver/payara-appserver-modules/microprofile/microprofile-common/src/main/java/fish/payara/microprofile/SetSecureMicroprofileConfigurationCommand.java

```diff
@@ -0,0 +1,108 @@
+ *     elects to include this software in this distribution under the [CDDL or GPL
+ *     Version 2] license."  If you don't indicate a single choice of license, a
+ *     recipient has the option to distribute your version of this file under
+ *     either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *     its licensees as provided above.  However, if you add GPL Version 2 code
+ *     and therefore, elected the GPL Version 2 license, then the option applies
+ *     only if the new code is made subject to such option by the copyright
+ *     holder.
+ */
+package fish.payara.microprofile;
+
+import static fish.payara.microprofile.Constants.DEFAULT_GROUP_NAME;
+import static fish.payara.microprofile.Constants.DEFAULT_USER_NAME;
+import javax.inject.Inject;
+import javax.security.auth.Subject;
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.CommandRunner;
+import org.glassfish.api.admin.ParameterMap;
+
+/**
+ *
+ * @author Gaurav Gupta
+ */
+public abstract class SetSecureMicroprofileConfigurationCommand implements AdminCommand {
+
+    @Param(optional = true, defaultValue = "server-config")
+    protected String target;
+    
+    @Inject
+    protected CommandRunner commandRunner;
+    
+    protected boolean defaultMicroprofileUserExists(ActionReport subActionReport, Subject subject) {
+        boolean exists = false;
+
+        CommandRunner.CommandInvocation invocation
+                = commandRunner.getCommandInvocation(
+                        "list-file-users",
+                        subActionReport,
+                        subject,
+                        false
+                );
+
+        ParameterMap parameters = new ParameterMap();
+        parameters.add("authrealmname", "file");
+
+        invocation.parameters(parameters).execute();
+
+        for (ActionReport.MessagePart message : subActionReport.getTopMessagePart().getChildren()) {
+            if (message.getMessage().equals(DEFAULT_USER_NAME)) {
+                exists = true;
+                break;
+            }
+        }
+
+        return exists;
+    }
+
+    protected void createDefaultMicroprofileUser(ActionReport subActionReport, Subject subject) {
+        CommandRunner.CommandInvocation invocation
+                = commandRunner.getCommandInvocation(
+                        "create-file-user",
+                        subActionReport,
+                        subject,
+                        false
+                );
+
+        ParameterMap parameters = new ParameterMap();
+        parameters.add("groups", DEFAULT_GROUP_NAME);
+        parameters.add("userpassword", "mp");
+        parameters.add("target", target);
+        parameters.add("authrealmname", "file");
+        parameters.add("DEFAULT", DEFAULT_USER_NAME);
+
+        invocation.parameters(parameters).execute();
```

Comment:
If the command fails, does it fail the parent command as well?

The user might have chosen not to use default realm, and even remove `file` from the realms altogether.

Reply:
Good catch, fixed it.